### PR TITLE
Added eth_getBlockByNumber and eth_getUncleByBlockNumberAndIndex

### PR DESCRIPTION
**Pull request by ntallar**
## Description

Includes `eth_getBlockByNumber` and `eth_getUncleByBlockNumberAndIndex` JSON RPC methods.

**Note:** Although according to https://github.com/ethereum/wiki/wiki/JSON-RPC#returns-26 `number`, `hash`, `nonce` and `logsBloom` should be `null` when the block tag is `Pending`, to keep the compatibility with geth the parameters selected as `null` were chosen as mentioned in https://github.com/ethereumproject/go-ethereum/blob/master/eth/api.go#L578.

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala [hunk 1]
```diff
@@ -198,10 +219,34 @@ class EthService(
...
+        if (uncleIndex >= 0 && uncleIndex < block.body.uncleNodesList.size)
+          Some(block.body.uncleNodesList.apply(uncleIndex.toInt))
+        else
+          None
+      }
+    val totalDifficulty = uncleHeaderOpt.flatMap(uncleHeader => blockchain.getTotalDifficultyByHash(uncleHeader.hash))
+
+    //The block in the response will not have any txs or uncles
+    val uncleBlockResponseOpt = uncleHeaderOpt.map { uncleHeader =>
+      BlockResponse(blockHeader = uncleHeader, totalDifficulty = totalDifficulty, pendingBlock = blockParam == BlockParam.Pending) }
```
Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala]: maybe `resolveBlock` should return information if block is pending
maybe something like
`val (uncleHeaderOpt, pending) = resolveBlock(blockParam)` ?

File: src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala [hunk 2]
```diff
@@ -157,11 +163,26 @@ class EthService(
...
+    * Implements the eth_getBlockByNumber method that fetches a requested block.
+    *
+    * @param request with the block requested (by it's number or by tag)
+    * @return the block requested or None if the client doesn't have the block
+    */
+  def getBlockByNumber(request: BlockByNumberRequest): ServiceResponse[BlockByNumberResponse] = Future {
+    val BlockByNumberRequest(blockParam, fullTxs) = request
+    val blockResponseOpt = resolveBlock(blockParam).toOption.map { block =>
+      val totalDifficulty = blockchain.getTotalDifficultyByHash(block.header.hash)
+      BlockResponse(block, totalDifficulty, fullTxs = fullTxs, pendingBlock = blockParam == BlockParam.Pending)
```
Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala]: same as in line 246

File: src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala [hunk 3]
```diff
@@ -105,6 +105,70 @@ class EthServiceSpec extends FlatSpec with Matchers with ScalaFutures with MockF
     response.transactionResponse shouldBe Some(expectedTxResponse)
   }
 
+  //FIXME: This test should be changed once the pending block doesn't equal to the latest one
```
Comment (adamsmo) [inline src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala]: should we address it now?

Comment (ntallar) [inline src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala]: We should probably address it soon (there's currently an open task for that https://iohk.myjetbrains.com/youtrack/issue/EC-184) but I think it will be better to handle this in a separate PR.

Comment (adamsmo) [inline src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala]: 👍 added comment to that task
